### PR TITLE
Unify Pull API documentation and remove deprectated restrictions

### PR DESCRIPTION
**Pull request by UliKuch**
This PR unifies the documentation of our Pull API into one document. The previous files now simply link to the new one. The Azubi-specific fields (which are the only platform-specific ones) are properly highlighted as such.

I also introduced some slight adjustments (see individual commits):
- Remove some explanatory text claiming that some fields were premium-specific, which is no longer the case. (I also adjusted the REST API documentation here.)
- Add missing explanation for the `custom_company_title` field.
- Adjust the application method explanation to the explanation we previously only had on Absolventa.
- Change the order of some fields is the data structure and in the explanation to ensure consistency and adjust the order to how these fields are displayed in the published job offers.

### CODE REVIEW

File: pull_api.md [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+        <p>Marley & Threepwood Ventures - Ihr Partner in Sachen Seefahrt- und Handelsbeziehungen</p>
+      ]]>
+    </company_description>
+
+    <video_url><![CDATA[https://api.example.com/videos/seemannsgarn]]></video_url>
+    <header_image_url><![CDATA[https://api.example.com/content/header_image.jpg]]></header_image_url>
+    <color>#efefef</color>
+    <custom_company_title>Threepwood GmbH</custom_company_title>
+
+    <!--attributes only relevant for azubi.de:  -->
```
Comment (dansch) [inline pull_api.md]:
```suggestion
    <!-- attributes only relevant for azubi.de:  -->
```

File: pull_api.md [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+    <apprenticeship_started_at type='datetime'>
+      2017-07-07T00:00:00+02:00
+    </apprenticeship_started_at>
+    <minimal_degree>basic</minimal_degree>
+    <duration>36</duration>
```
Comment (dansch) [inline pull_api.md:116]: Maybe just for clarification we can add an comment about the end of the azubi specific attributes?

**Review by dansch — APPROVED** [review]
👍 Nice simplification. I just left two minor hints.

File: pull_api.md (hunk 2 above)
Comment (UliKuch) [reply pull_api.md:116]: Good point, I added a second comment.

**Review by abbyhudson13 — APPROVED** [review]
👍

**Pull request merged**